### PR TITLE
style(form-field): fix height and spacing of preffix / suffix icon bu…

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description
fix height and spacing of preffix / suffix icon buttons in form-fields

## References

- #152

### CODE REVIEW

File: libs/ui-watt/src/lib/components/form-field/form-field.component.scss [hunk 1]
```diff
@@ -78,12 +85,27 @@ watt-form-field {
...
+    .watt-icon-button {
+      height: 40px;
+    }
+  }
+
   [wattPrefix] {
     margin-right: 8px;
+
+    .watt-icon-button {
+      margin-left: -12px;
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/form-field/form-field.component.scss:99]: issue: this is applying height to icons inside a `watt-icon-button` component. Often the prefix icon is a `watt-icon` component. Consider that the icon might be inside a `watt-icon-button` component. 

question: Will there be a case where the prefix icon is a `watt-icon-button`?

File: libs/ui-watt/src/lib/components/form-field/form-field.component.scss [hunk 2]
```diff
@@ -78,12 +85,27 @@ watt-form-field {
...
+    }
   }
 
   [wattSuffix] {
     margin-left: 8px;
+
+    .watt-icon-button {
+      margin-right: -12px;
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/form-field/form-field.component.scss:107]: question: Wouldn't it be better to move the whole component 12px to the right instead of just the button? Right now the button inside the component is moved to the right and when inspected, the component looks like this:

image

File: libs/ui-watt/src/lib/components/form-field/form-field.component.scss [hunk 3]
```diff
@@ -31,6 +31,13 @@ watt-form-field {
     .mat-form-field-flex {
       height: 56px;
     }
+
+    [wattPrefix],
+    [wattSuffix] {
+      .watt-icon-button {
+        height: 52px;
```
Comment (dzhavat) [inline libs/ui-watt/src/lib/components/form-field/form-field.component.scss:38]: issue: this is applying height to icons inside a `watt-icon-button` component. Often the prefix icon is a `watt-icon` component (e.g. inside `dh-metering-point-search-form.component.html`). This style won't be applied in that case. Will there be a case where the prefix icon is a `watt-icon-button`?

File: libs/ui-watt/src/lib/components/form-field/form-field.component.scss [hunk 4]
```diff
@@ -78,12 +85,27 @@ watt-form-field {
     display: none;
   }
 
+  [wattPrefix],
+  [wattSuffix] {
+    .watt-icon-button {
+      height: 40px;
```
[the remark this answers is not in the document]
Comment (frontend-specialisten) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss:91]: The search form is also using a `watt-icon-button` when the icon is clickable (the reset). If it's just an indicator like the search icon, it won't have this issue as it's not clickable and won't get any outline (it's not even focusable). This is why I'm only targeting the `watt-icon-button` :)

File: libs/ui-watt/src/lib/components/form-field/form-field.component.scss (hunk 2 above)
Comment (frontend-specialisten) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss:107]: If we're targeting the whole `suffix`, icons would also be moved because it doesn't have any padding.

Comment (dzhavat) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss:107]: What about targeting `watt-icon-button` elements with `wattSuffix` e.g. `watt-icon-button[wattSuffix]`?

Comment (frontend-specialisten) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss:107]: Unfortunately `watt-icon-button[wattSuffix]` won't work as this won't change the height of the button itself.

Comment (dzhavat) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss:107]: This was more related to the `margin-right`. Instead of moving the inner button, we can move the whole `watt-icon-button` element. Wouldn't that work?

Comment (frontend-specialisten) [reply libs/ui-watt/src/lib/components/form-field/form-field.component.scss]: Selector regarding margin-right and left has now been changed

File: .github/workflows/frontend-ci.yml [hunk 5]
```diff
@@ -379,12 +373,6 @@ jobs:
       - name: Install package dependencies
         run: yarn install --frozen-lockfile --non-interactive
 
-      - name: 'DH: Start API'
-        if: ${{ env.is-main-branch == 'true' || needs.dh-affected.outputs.is-affected == 'true' }}
-        run: |
-          yarn nx run api-dh:ci-configuration
-          yarn nx serve api-dh:serve:ci
-
```
Comment (dzhavat) [inline .github/workflows/frontend-ci.yml:387]: question: Are these changes necessary in this PR?

Comment (frontend-specialisten) [reply .github/workflows/frontend-ci.yml]: No that was a mistake, changes has now been reverted

**Review by dzhavat — APPROVED** [review]
